lib/apex-class-cleanup.js

```diff
@@ -27,6 +28,7 @@ let login = function(options) {
     let connection = new jsforce.Connection(config);
     return connection
         .login(options.user, options.password + options.securityToken)
+        .then(connection.metadata.pollTimeout = options.pollTimeout || FallbackPollTimeout)
```

Comment:
remove line. this code actually runs before login because this is a statement not a callback. the statement gets evaluated before it is passed into the then function. so the poll timeout in ms is being send as a promise callback. as in, this executes as 
.then(100000) after 100000 is assigned to connection.metadata.pollTimeout

Reply:
@bruce-lindsay weird. i debugged the connection prior to adding the "then," and after, to ensure that the interval was updating correctly (just two console.logs before and after this ternary) and it had correctly updated. i guess i assumed that all of the then's would get evaluated in order due to the promise chain.  more than happy to make the changes!  i have to run to the dentist but i'll get back to this afterwards - thank you for your feedback!

Reply:
okay, maybe there is something going on i can't recognize

Reply:
You are correct. Okay please ignore my feedback on this. Apparently the code in the parameter list is automatically turned into a function and called at the appropriate time.
Tested using this code... 
https://runkit.com/bruce-lindsay/5a9049ce9f37ad0012b3c6f4

Reply:
@bruce-lindsay whew. I thought I was going crazy for a second.  Runkit is really cool, thanks for the link!